### DISCUSSION TITLE
2022-09-09-miaoyan

### DISCUSSION THREAD

**Discussion by giscus** [tw93/tw93.github.io#113] [General]
# 妙言 - 更适合工程师用的笔记应用 - Tw93

Hi，我是 Tw93，是一个前端工程师，会一点 Swift / Rust / Python，喜欢折腾各种新技术。Tool这篇文章主要想给大家介绍一下妙言的功能清单，为啥更适合工程师使用，也包括个人对于产品本身的一些想法。

https://tw93.fun/2022-09-09/miaoyan.html

**Comment by elizen** [comment]
这个好帅啊。有几个功能很有心了。

**Reply by whsheng** [reply]
我是从fsnotes了解「妙言」，如作者所言，希望在Apple生态里寻找一款安全、简单、不依赖的笔记应用！目前来看，妙言在某些方面要比fsnotes做的要好，加油！

但也希望后续有类似功能：文件加密、标签

**Comment by 0xtulaoda** [comment]
大佬，妙言有计划支持保存数据到iCloud么

**Reply by tw93** [reply]
现在就支持的，你可以试试，直接在 iCloud 里面建立文件夹即可

**Reply by kamusis** [reply]
在macOS下使用很顺手，文件直接存在iCloud里面也非常方便。不确认是不是同样完美支持OneDrive等云端存储，理论上似乎是直接指定OneDrive的文件路径给妙言就行。但是实际上，以前在使用Obsidian等app的时候，发现，如果文档存储在OneDrive中，自动保存的时候会有延迟，甚至产生莫名其妙的丢字等现象。
OneDrive能够更好地支持Windows，Linux，macOS三者的跨平台文档同步体验，而iCloud就通常只在Apple生态下表现良好。

**Reply by tw93** [reply]
支持的 只要可以读取到 就行

**Comment by lleoiu** [comment]
可以期待一下妙言的Windows版本吗？

**Reply by SevDaisy** [reply]
自己fork一个，迁移一下？

**Reply by hezhan9986** [reply]
我也在期待（虽然大佬已经表示目前暂时没有这一计划）

**Comment by harleyszhang** [comment]
行间公式是不是支持的不好，我这里使用显示不了。

**Reply by tw93** [reply]
可以去issue里面看看解法，比如说用 `<span>` 或者`<div>`包裹下

**Comment by Stellanliu** [comment]
有没有考虑给sidebar folder加个层级关系，希望有个树状层级列表

**Reply by Stellanliu** [reply]
顺便希望给**编辑模式下**的LaTeX输入公式加个高亮颜色，这样可以和正常文本区分开来，e.g. $\vec{g}=\vec{a}\times\vec{b}$
感觉提了挺多需求了，希望别给作者太多压力:)

**Comment by ImpTyrion** [comment]
1. 不能很好的在左侧显示文件目录
2. 希望支持一下文件排序(例如: 3th.md 5th.md 1th.md 自动排序后为 1th.md 3th.md 5th.md)
3. 左侧也不够显示文件夹目录，无论如何我怎么操作，比如我把目录拖进妙言，是不支持的
ps: 评论去貌似无法粘贴截图(例如: input textare 可换成  div设置contenteditable 实现)

**Comment by tianzhiqiang1110** [comment]
没办法展示图片能插入图片就好啦

**Reply by tw93** [reply]
可以试试 `![](图片地址的方式)`

**Comment by 0xkevin** [comment]
请问怎么插入本地图片？为什么用markdown方式插入图片会不识别？这是一个bug？

**Reply by tw93** [reply]
可以试试直接拖进去，或者用command+v复制的方式，都可以，不过建议使用 `![](图片CDN地址的方式)`

**Comment by zsmdzt** [comment]
你好，不好意思，由于本人小白，看中Markmap功能，请问怎么使用呢

**Reply by tw93** [reply]
```
```markmap
里面写markdown即可
```
```

**Comment by KaiyoungYu** [comment]
昨天在寻找obsidian适合中文的主题，之前一直用的everforest 有点审美疲劳了。刚才看到妙言的黑色主题觉得十分合适，我打算抄袭一下你的配色，自己配置一个主题出来。哈哈哈。博主会介意吗？

**Reply by tw93** [reply]
没事的

**Comment by hezhan9986** [comment]
作为一名win用户，看着妙言垂涎三尺！😂

**Comment by Andark** [comment]
妙言，给我感觉就是干净，没有其他花哨的功能“污染”，因而更聚焦于它的本源。这一点很好。另外，着实喜欢妙言的中文字体。不知道“即时编辑”什么时候发布，我觉得这个应该会带来更好的体验。很期待。

**Comment by Abrygzy** [comment]
希望可以增加大纲功能, 比如在左边栏展示出文章中的标题

**Comment by lu-men** [comment]
大佬您好，请教一下，ppt模式下文件能否导出呢？似乎现在导出只有pdf格式，如果能导出分页的pdf文件直接去展示，那就太好了，谢谢！

**Comment by jaywhj** [comment]
你好，我是今天才看到妙言，试用了一下，给里面的PPT模式惊艳到了，想迁移过来，但是又有几处还不是还很完善，如果能完善一下就更好了，我就完全可以抛弃typora了😄，可以参考下：

1. 本地路径不识别，在插入本地资源过程中，`![本地图片](/Users/aaron/Pictures/md-mty.jpg)` 加载不出来；
2. 跳转标题有bug，标题中间如果有空格，则跳不过去。比如我的标题是`1. 标题`，写成 `[显示文字](#1. 标题)` 跳不过去；
3. 功能区-大纲视图，typora最方便的功能，也是我使用typora的最主要原因，方便跳转导航，使用参考typora；
4. 文档内大纲视图，即[TOC]目录支持，这也是一个很方便的功能；
5. 编辑模式下，不支持内容搜索及跳转，不是很方便；
6. 缺一些特性的使用文档或说明。我看说明支持流程图、序列图、甘特图，但是试了一下，都没加载出来，从mermaid复制code过来的；

**Comment by jaywhj** [comment]
除了上面的，再补充几个小bug：
1. 内联代码里的信息会被解析，而原本是用来展示代码的。比如 用 `:smile:` 表示 :smile: ，``里的表情在编辑器里会解析出来，而原本应该显示字符代码的。
2. 编辑器中的音视频没法播放，能显示，应该是没添加解析器支持。
3. 编辑功能体验还是不够，等待即使编辑上线，还是大纲视图方便😂。
4. 数学公式和HTML展示都没问题，但是图表功能不知道怎么用，测试了官方代码和其他编辑器流行写法都加载不出来。
5. 导出功能还不是很完善，导出PDF分页机制和留边不正常。

整体来说，已经很棒了，bug 比 MWeb 少多了，已经是最接近Typora的一款新型编辑器了👍，简洁方便，而特色功能PPT模式更是很惊艳，非常期待作者能再修复一版本啦，加油加油！

**Comment by Jie-Huangi** [comment]
你好，windows还会考虑吗？

**Comment by lasolzol** [comment]
今天刚试用，确实很惊艳。
不过发现两个问题：
1.中文输入法状态下，输入英文，回车后，应该默认接受英文的，但回车后会吞掉输入，即：没有接受到输入；
2.我在快速跳转点击默认一些介绍文档后，突然卡顿了一下，之后所有文档都被清空了，包括我刚建的一个文档，不清楚原因和机制。
